client/app/views/menu_item.coffee

```diff
@@ -29,6 +25,16 @@ module.exports = class MenuItemView extends BaseView
     afterRender: ->
         @buildBadge @model.get 'color'
 
+    toggleVisible: ->
+        unless app.router.onCalendar
+            app.router.navigate 'calendar', true
+        @startSpinner()
+        setTimeout => # async, to allow spinner to show up.
+                @model.set 'visible', not @model.get 'visible'
+                @stopSpinner()
+                @render()
+            , 1
```

Comment:
That's why I have a strange behaviour: you basically say "start the spinner, wait 1ms, stop the spinner" so it logically blinks.